packages/discovery-react-components/src/components/SearchFacets/SearchFacets.tsx

```diff
@@ -143,8 +146,15 @@ const SearchFacets: FC<SearchFacetsProps> = ({
       }
     }
 
-    if (searchParamsAggregationChanged || fetchState === 'init') {
+    if (
+      (searchParamsAggregationChanged || fetchState === 'init') &&
+      !isReturningFromResultPreview
+    ) {
       fetchData();
+    } else {
+      // we don't need to refetch data if returning to search results and facets from previewing a result
+      setIsReturningFromResultPreview(false);
```

Comment:
im struggling to understand why this is necessary - is `fetchState` being reset to `init` when clicking on a search result? is there any way we could move the initial `fetchState` outside of this component?

Reply:
Yes exactly--`fetchState` was being reset to `init` when previewing a search result and then returning to search results/facets. I moved the `fetchState` out of this component into `DiscoverySearch` instead and like this approach better, thanks! 👍